### DISCUSSION TITLE
Where is the source css code of the three @tailwind directives

### DISCUSSION THREAD

**Discussion by yf-yang** [tailwindlabs/tailwindcss#12705] [Help] [answered]
I've found https://github.com/tailwindlabs/discuss/issues/397, but still can't figure it out.
I'm expecting some file(s) containing lines of css codes that matches `@tailwind base` and other two directives. Where are they?

**Comment by wongjn** [comment, marked as answer]
If you are asking for how the `@tailwind` directives are processed into CSS rules, the source code for that is at https://github.com/tailwindlabs/tailwindcss/blob/8350cffdad34efb6625e0f2b0ff738edaf0d1cce/src/lib/expandTailwindAtRules.js#L100-L300

If you are asking for CSS files that contain `@tailwind` directives, then there are:
- [`base.css`](https://github.com/tailwindlabs/tailwindcss/blob/v3.4.0/base.css)
- [`components.css`](https://github.com/tailwindlabs/tailwindcss/blob/v3.4.0/components.css)
- [`screens.css`](https://github.com/tailwindlabs/tailwindcss/blob/v3.4.0/screens.css)
- [`tailwind.css`](https://github.com/tailwindlabs/tailwindcss/blob/v3.4.0/tailwind.css)
- [`utilities.css`](https://github.com/tailwindlabs/tailwindcss/blob/v3.4.0/utilities.css)
- [`variants.css`](https://github.com/tailwindlabs/tailwindcss/blob/v3.4.0/variants.css)

**Reply by yf-yang** [reply]
@wongjn Thanks! Let me clarify my question:
If I got it right, those `@tailwind xxx` will be transpiled/converted to real css code by postcss. Therefore, I would like to know **what css code will be generated** when I type `@tailwind xxx`. In the `*.css` file you listed, all of them contains only the `@tailwind xxx` directives.

**Reply by wongjn** [reply]
Thank you for the clarification. The CSS code that will be generated for the directives is processed by:

https://github.com/tailwindlabs/tailwindcss/blob/8350cffdad34efb6625e0f2b0ff738edaf0d1cce/src/lib/expandTailwindAtRules.js#L100-L300

**Reply by yf-yang** [reply]
🤔 Seems its from https://github.com/tailwindlabs/tailwindcss/blob/47107be7835b41f2771ceddd314a34a0a85abe64/src/css/preflight.css. Is it the only source file?

**Reply by wongjn** [reply]
That is for the [Preflight core Tailwind plugin](https://tailwindcss.com/docs/preflight). The CSS rules are generated from the source code as detailed at https://github.com/tailwindlabs/tailwindcss/discussions/12705#discussioncomment-8010429.
